sonar/inference_pipelines/text.py

```diff
@@ -165,12 +166,19 @@ def truncate(x: torch.Tensor) -> torch.Tensor:
                 nonlocal n_truncated
                 n_truncated += 1
             return x[:max_seq_len]
+        
+        def sort_input(input_sent) -> Sequence[str]:
+            return sorted(input_sent, key=len) if sort_sent else input_sent
+
+        if isinstance(input, (str, Path)):
+            input_data = read_sequence(sort_input(read_text(input)))
+            input_data = read_text(input).and_return()
+        else:
+            input_data = read_sequence(sort_input(input))
 
         pipeline: Iterable = (
             (
-                read_text(input)
-                if isinstance(input, (str, Path))
-                else read_sequence(input)
+                input_data
```

Comment:
at the end we want to return vectors in the same order as their original texts, so we need to memorize `argsort` permutation and to apply the inverse one before returning the result.